src/define.h

```diff
@@ -0,0 +1,10 @@
+#ifndef DEFINE_H
+#define DEFINE_H
+
+
+#define DIM 2 // dimension of the problem to solve
+#define SKIPS 1 // Number of output time steps to skip
+#define LEAP_FROG false
+#define CONVERGENCE false // Performs convergence analysis
```

Comment:
Same as before on preprocessor directives.

To articulate a little better why that is, here you could e.g. write by mistake
```cpp
#define LEAP_FROG "false"
```
or any other non-boolean expression, leading to relatively hard to debug errors. Even worse, you might be tempted to write something like
```cpp
#define CONVERGENCE DIM == 2
```
which may (or may not) lead to incorrect expansions such as
```cpp
if (CONVERGENCE == true) -> if (DIM == 2 == true)
```

Conversely, more modern approaches to the same goal, like `if constexpr` statements, template specialization and other metaprogramming techniques, are usually more robust and overall better.

For a few of these flags, moreover, a run-time variable might have worked just fine.